src/Math-Matrix/PMMatrix.class.st

```diff
@@ -588,6 +588,17 @@ PMMatrix >> orthogonalize [
 
 ]
 
+{ #category : #'as yet unclassified' }
+PMMatrix >> principalDiagonal [
+	"https://en.wikipedia.org/wiki/Diagonal#Matrices for definitions"
+	| diag |
+	"Check for square"
+	self isSquare ifFalse: [ self error: 'Diagonal is not defined for a matrix that is not square.' ].
+	diag := PMVector new: self rows size.
```

Comment:
Here you could use the `numberOfColumns` message (pretty sure that exists) to be a little more direct. What do you think?

Reply:
I agree with both your comments. However, I'm not sure when I'll have time to make these changes in an image and re-commit them.

Reply:
I can make the changes in another PR, no problem at all.